sklearn/impute/_base.py

```diff
@@ -667,7 +663,31 @@ def fit_transform(self, X, y=None):
             will be boolean.
 
         """
-        return self.fit(X, y).transform(X)
+
+        """fit"""
+        X = self._validate_input(X)
+        self._n_features = X.shape[1]
+
+        if self.features not in ('missing-only', 'all'):
+            raise ValueError("'features' has to be either 'missing-only' or "
```

Comment:
You shouldn't duplicate the code ideally. It's not covered by the tests which is why you're seeing the code coverage failure.

Reply:
Then what should I do, because I need to do the exact same things, I can't change the variable name (can I??). I can't call fit function because if I do I need to create a data member to store what the function returns. 
Any help??

Thanks.

Reply:
@amueller Then what should I do, because I need to do the exact same things, I can't change the variable name (can I??). I can't call fit function because if I do I need to create a data member to store what the function returns which raises more failures during check because the tests do not know have any information about it. 
Any help??

Thanks.

Reply:
@amuller Can I use mangling??